### DISCUSSION TITLE
Why ClusterTopologyRefreshOptions is off by default

### DISCUSSION THREAD

**Discussion by canmanciu** [redis/lettuce#2814] [Q&A] [answered]
Why is ClusterTopologyRefreshOptions turned off by default? Developers who are not familiar with lettuce may not realize that they need to manually turn it on, resulting in failure to fail over. I would like to know what the considerations are. Are there any performance issues to consider?

**Comment by mp911de** [comment, marked as answer]
This is because the library cannot reason about changes in your runtime environment, any periodic schedule and other environmental aspects that are required to know what type of topology refresh might make sense.

**Reply by canmanciu** [reply]
Would it be more reasonable to start scheduled tasks as needed based on specific environmental judgments ?

**Reply by tishun** [reply]
Could you suggest a solution for these defaults?
